src/LEGO.AsyncAPI.Bindings/Sns/SnsChannelBinding.cs

```diff
@@ -0,0 +1,121 @@
+using System.Linq;
+using LEGO.AsyncAPI.Bindings;
+using LEGO.AsyncAPI.Models;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+using LEGO.AsyncAPI.Writers;
+using YamlDotNet.Core.Tokens;
+
+namespace LEGO.AsyncAPI.Bindings.Sns;
+
+using LEGO.AsyncAPI.Models.Interfaces;
+
+/// <summary>
+/// Binding class for SNS channel settings.
+/// </summary>
+public class SnsChannelBinding : ChannelBinding<SnsChannelBinding>
+{
+    /// <summary>
+    /// The name of the topic. Can be different from the channel name to allow flexibility around AWS resource naming limitations.
+    /// </summary>
+    public string Name { get; set; }
+
+    /// <summary>
+    /// By default, we assume an unordered SNS topic. This field allows configuration of a FIFO SNS Topic.
+    /// </summary>
+    public OrderingConfiguration Type { get; set; }
+
+    /// <summary>
+    /// The security policy for the SNS Topic.
+    /// </summary>
+    public Policy Policy { get; set; }
+
+    /// <summary>
+    /// Key-value pairs that represent AWS tags on the topic.
+    /// </summary>
+    public Dictionary<string, string> Tags { get; set; }
+
+    public override string BindingKey => "sns";
+
+    protected override FixedFieldMap<SnsChannelBinding> FixedFieldMap => new()
+    {
+        { "name", (a, n) => { a.Name = n.GetScalarValue(); } },
+        { "type", (a, n) => { a.Type = LoadType(n); } },
+        { "policy", (a, n) => { a.Policy = LoadPolicy(n); } },
+        { "tags", (a, n) => { a.Tags = n.CreateSimpleMap(s => s.GetScalarValue()); } },
+    };
+
+    private static FixedFieldMap<OrderingConfiguration> orderingFixedFields = new()
+    {
+        { "type", (a, n) => { a.Type = n.GetScalarValue().GetEnumFromDisplayName<Ordering>(); } },
+        { "contentBasedDeduplication", (a, n) => { a.ContentBasedDeduplication = n.GetBooleanValue(); } },
+    };
+
+    private static FixedFieldMap<Policy> policyFixedFields = new()
+    {
+        { "statements", (a, n) => { a.Statements = n.CreateList(s => LoadStatement(s)); } },
+    };
+
+    private static FixedFieldMap<Statement> statementFixedFields = new()
+    {
+        { "effect", (a, n) => { a.Effect = n.GetScalarValue().GetEnumFromDisplayName<Effect>(); } },
+        { "principal", (a, n) => { a.Principal = LoadStringOrStringList(n, "principal"); } },
+        { "action", (a, n) => { a.Action = LoadStringOrStringList(n, "action"); } },
+    };
+
+    private static OrderingConfiguration LoadType(ParseNode node)
+    {
+        var mapNode = node.CheckMapNode("type");
+        var ordering = new OrderingConfiguration();
+        ParseMap(mapNode, ordering, orderingFixedFields);
+        return ordering;
+    }
+
+    private static Policy LoadPolicy(ParseNode node)
+    {
+        var mapNode = node.CheckMapNode("policy");
+        var policy = new Policy();
+        ParseMap(mapNode, policy, policyFixedFields);
+        return policy;
+    }
+    
+    private static Statement LoadStatement(ParseNode node)
```

Comment:
note, ive made a small qol improvement.

Moving parsemap to be an extension method on ParseNode instead.

So you can use it directly in the FixedFieldMap instead of making these methods forcing a mapnode etc.

```csharp
protected override FixedFieldMap<PulsarChannelBinding> FixedFieldMap => new()
        {
            { "bindingVersion", (a, n) => { a.BindingVersion = n.GetScalarValue(); } },
            { "namespace", (a, n) => { a.Namespace = n.GetScalarValue(); } },
            { "persistence", (a, n) => { a.Persistence = n.GetScalarValue().GetEnumFromDisplayName<Persistence>(); } },
            { "compaction", (a, n) => { a.Compaction = n.GetIntegerValue(); } },
            { "retention", (a, n) => { a.Retention = n.ParseMap("rentention", this.pulsarServerBindingRetentionFixedFields); } }, // HERE
            { "geo-replication", (a, n) => { a.GeoReplication = n.CreateSimpleList(s => s.GetScalarValue()); } },
            { "ttl", (a, n) => { a.TTL = n.GetIntegerValue(); } },
            { "deduplication", (a, n) => { a.Deduplication = n.GetBooleanValue(); } },
        };
```